### DISCUSSION TITLE
Unable to find custom connector using Terraform

### DISCUSSION THREAD

**Discussion by anzelpwj-rev** [airbytehq/airbyte#40699] [Connector Questions] [answered]
Hello everyone,

I've been struggling over the past few weeks trying to get terraform to properly create a connector based on something I've built in the low-code builder. I went to the low-code builder, copied the URL (`http://localhost:8000/workspaces/86835965-503b-43b6-ab6b-c25e0d606f11/connector-builder/edit/10035bb3-1f50-4f34-acbe-c412803cddfc`). and then created a terraform step like:

```
resource "airbyte_source_custom" "source_impact" {
  configuration = jsonencode(
    { "username" : var.impact_username,
      "password" : var.impact_password }
  )
  name          = "Impact"
  workspace_id  = "86835965-503b-43b6-ab6b-c25e0d606f11"
  definition_id = "10035bb3-1f50-4f34-acbe-c412803cddfc"
}
```

Terraform plan works, but when I run terraform apply I get the following error:

```

│ Error: unexpected response from API. Got an unexpected response code 400
│
│   with airbyte_source_custom.source_impact,
│   on main.tf line 16, in resource "airbyte_source_custom" "source_impact":
│   16: resource "airbyte_source_custom" "source_impact" {
│
│ **Request**:
│ POST /v1/sources HTTP/1.1
│ Host: host.docker.internal:8006
│ Accept: application/json
│ Authorization: (sensitive)
│ Content-Type: application/json
│ User-Agent: speakeasy-sdk/go 0.0.1 2.332.4 1.0.0 github.com/airbytehq/terraform-provider-airbyte/internal/sdk
│
│
│ **Response**:
│ HTTP/1.1 400 Bad Request
│ Content-Length: 197
│ Connection: keep-alive
│ Content-Type: application/problem+json
│ Date: Tue, 02 Jul 2024 15:25:18 GMT
│ Server: nginx/1.25.3
│
│ {"type":"https://reference.airbyte.com/reference/errors#resource-not-found","detail":"Could not find a resource for:
│ 86835965-503b-43b6-ab6b-c25e0d606f11","title":"resource-not-found","status":400}
```

If I manually create the source from the custom object and inspect it via the API I can see it clearly on `v1/sources`, but [`v1/source_definitions/list_for_workspace`](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/source_definitions/list_for_workspace) and [`v1/source_definitions/list`](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/source_definitions/list) are giving me `{"type":"about:blank","status":404}` errors.

Am I making some fundamental mistake?

**Comment by szemek** [comment, marked as answer]
@anzelpwj-rev it doesn't look as a good definition_id. Can you try get definition_id from database with query?

```sql
SELECT actor_definition_id AS definition_id
FROM connector_builder_project
WHERE id = '10035bb3-1f50-4f34-acbe-c412803cddfc';
```

**Reply by anzelpwj-rev** [reply]
Any advice on how to log into the Airbyte database? I tried following the instructions at https://docs.airbyte.com/operator-guides/configuring-airbyte-db#accessing-the-default-database-located-in-docker-airbyte-db and the actor and did not find anything in either the `actor` or `connector_builder_project`.

EDIT: NVM, I've been so used to other SQL engines that I forgot the closing semicolon.

**Reply by anzelpwj-rev** [reply]
Okay, so the UUIDs are in the `id` column, but not in the `actor_definition_id` columns. Should I be using that instead?

**Reply by anzelpwj-rev** [reply]
Okay, I just tested with using the `actor_definition_id` value instead of `id` for `connector_builder_project`, and that resolved the problem. Thank you for your help!
